sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -500,6 +503,18 @@ object LikeSimplification extends Rule[LogicalPlan] {
             Contains(input, Literal(infix))
           case equalTo(str) =>
             EqualTo(input, Literal(str))
+          // case: abc_
+          case endsWithUnderscore(prefix) =>
```

Comment:
what if the pattern is something like `abc\_`?

Reply:
I have handled this case and added UTs for this. Thanks!